python/engine/evaluation.py

```
@@ -13,14 +13,182 @@
+
+    return {
+        "run1_avg_loss": avg1,
+        "run2_avg_loss": avg2,
+        "improvement": improvement,
+        "improvement_pct": improvement_pct,
+        "run1_text": text1,
+        "run2_text": text2,
+    }
+
+
+def probe_context_value(gpu_model, cfg, prompt_ids, snapshot, tokenizer,
```

Ruff (0.15.2)

[warning] Unused function argument: `tokenizer`

(ARG001)